app/models/manageiq/providers/amazon/cloud_manager.rb

```diff
@@ -119,6 +119,18 @@ def self.default_blacklisted_event_names
     Settings.ems.ems_amazon.blacklisted_event_names
   end
 
+  def self.params_for_create
+    {
+      :title  => "Configure AWS",
+      :fields => [
+        {:component => "text-field", :name => "role",         :type => "hidden", :initialValue => "ec2"},
+        {:component => "text-field", :name => "region",       :label => "Region"},
+        {:component => "text-field", :name => "ec2_username", :label => "Access Key"},
+        {:component => "text-field", :name => "ec2_password", :label => "Secret Key", :type => "password"}
+      ]
```

Comment:
Having this flat structure might not be enough if you're expecting the data in the validation method in a nested form. Here all the items in the DDF schema are on the same level, but below you wrap everything except of `region` around an `endpoint`.

Reply:
For nested values you can write something like `endpoints.ec2.username` and DDF will send it as 
```jsx
endpoints: {
   ec2: {
      username: `myname`
 }
}
```

Reply:
Okay great, yeah I'm not very familiar with DDF so was looking to you guys for how to get the structure back that we're looking for.

Reply:
Okay how does that look?

Reply:
Nesting like this is great. We can go even step further and add `authentication.default.userid` to match the actual API request payload (if there is any). But we can add prefixes easily later on.

Reply:
Okay there's only one endpoint so we can just flatten out the structure, good to know how to nest under endpoints for providers like ovirt though :+1: